src/libraries/Common/src/System/Net/IPv4AddressHelper.Common.cs

```diff
@@ -2,32 +2,40 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Numerics;
 
-namespace System
+namespace System.Net
 {
-    internal static partial class IPv4AddressHelper
+    internal static partial class IPv4AddressHelper<TChar>
+            where TChar : unmanaged, IBinaryInteger<TChar>
     {
+        // IPv4 address-specific generic constants.
+        public static readonly TChar ComponentSeparator = TChar.CreateTruncating('.');
+        public static readonly TChar[] PrefixSeparators = [TChar.CreateTruncating('/'), TChar.CreateTruncating('\\')];
+        public static readonly TChar[] UrlSeparators = [TChar.CreateTruncating(':'), TChar.CreateTruncating('?'), TChar.CreateTruncating('#')];
+        public static readonly TChar[] HexadecimalPrefix = [TChar.CreateTruncating('0'), TChar.CreateTruncating('x')];
+        public static readonly TChar OctalPrefix = HexadecimalPrefix[0];
```

Comment:
Same as above, I'd keep these defined inline.

Checking `ch == TChar.CreateChecked('/') || ch == TChar.CreateChecked('\\')` is also a lot more efficient than `Array.IndexOf(PrefixSeparators, ch) != -1`